freelist.go

```diff
@@ -171,7 +174,11 @@ func (f *freelist) read(p *page) {
 	copy(f.ids, ids)
 
 	// Make sure they're sorted.
-	sort.Sort(pgids(f.ids))
+	for i, v := range f.ids[:len(f.ids)-1] {
```

Comment:
That is true, although it does matter how many of those integers are copied around. Those copies are not necessary as far as I can see and can be avoided. It's similar to how in this PR we are proposing that bool values are more expensive compared to empty struct{}s, when in fact allocation for individual bools on a small scale is supposedly very cheap (on the order of a byte).

I think the assumption of function calls being expensive in Go doesn't necessarily stand true. The compiler inlines a ton of functions behind the scenes. Unless we can write a benchmark that proves calling a function (that only does a integer comparison) is indeed that expensive I think we should default to using the ones provided by stdlibs.

If we are worried about performance I think we should lean towards `sort.IsSorted` or at least prove this implementation is equivalent. However, If we don't care about it too much we can proceed with this implementation.